### ISSUE TITLE
Information on using just the API

### ISSUE THREAD

**Issue by damontgomery** [cucumber/aruba#458]
## Summary

I'd like to know how to run the API commands directly in case the steps provided by cucumber/aruba aren't enough and / or I want to write them in stakeholder friendly ways. This is a pattern Capybara uses. It's easy to get at all of their methods for visiting pages, etc.

I'm using Cucumber to run the tests.

I've tried to copy the pattern here, https://github.com/cucumber/aruba/blob/master/lib/aruba/cucumber.rb

And added this,

```
require 'aruba/version'

require 'aruba/api'
World(Aruba::Api)
```

However, my step definitions can't find any of the methods.

Example:

```
run_command_and_stop("ls", :fail_on_error => false)
```

## Your Environment

* Version used: 0.14.2
* Operating System and version: OSX, but tried in Ubuntu as well.

**Comment by maxmeyer** [comment]
@damontgomery Sorry for the inconvinience. The docs on master are for `1.0.0-alpha.2` currently. What you want is `run_simple` and have a look into https://github.com/cucumber/aruba/tree/still.

**Comment by damontgomery** [comment]
Don't be sorry!

It looks like I should try to use the 1.0.0-alpha.2 branch to be aligned with the docs. That way, I can use the commands I'm seeing. If I want to keep using 0.14.2, I can find the equivalent functions in the links you provided.  See in `https://github.com/cucumber/aruba/blob/still/lib/aruba/cucumber/command.rb`.

Would it be worth adding a section on the README about how to create your own step definitions? I'm new to Cucumber, Ruby, etc, so I'm probably not familiar with a lot of the concepts. The Capybara docs again are fairly easy to follow because I can create my own step definitions and see which methods are available. Maybe this is the equivalent, http://www.rubydoc.info/github/cucumber/aruba/master/Aruba/Api/Commands.

See Capybara, http://www.rubydoc.info/github/teamcapybara/capybara/master#the-dsl

I know it's a lot of work to write documentation.

Can you add some more context to the branches? What is the intent of the `still` branch? It looks older than the master branch. It looks like there are a few small changes beyond master, but a lot behind, https://github.com/cucumber/aruba/compare/still.

Thank you!

**Comment by maxmeyer** [comment]
> Maybe this is the equivalent, http://www.rubydoc.info/github/cucumber/aruba/master/Aruba/Api/Commands.

Yes, that's a place to look for documentation, but I would recommend to have a look at our generated documention on Cucumber.Pro (https://app.cucumber.pro/projects/aruba). This should be easier to work with and it also offers a search engine. If you look for "aruba api" you should find all information you need. If not, just answer to this issue.

> Would it be worth adding a section on the README about how to create your own step definitions?

I think that's out of scope of "aruba"'s README. Instead I added some links to the documentation websites of all supported testing frameworks. This should make it easier to find their documentation.

BTW: How do you look for documentation? Google?

I'm going to close this for now, but feel free to answer to this issue.

**Comment by damontgomery** [comment]
Thanks!

I was able to use the methods in the examples on Cucumber Pro now that I'm using the Alpha 2 release.

In case it helps someone else,

In `features/support/aruba.rb` I have,
```
require 'aruba/cucumber'
```

per the documentation. This provides access to the Aruba API as well as step definitions. 

I then made some steps like,

```
Then(/^I should get the status of the example application$/) do
  run_command("example-application status")
  expect(last_command_started).to be_successfully_executed
end

Then(/^I should not get the status of the example application$/) do
  run_command("example-application status")
  expect(last_command_started).not_to be_successfully_executed
end
```

About your question on how to find information,

I started with the Github page, then looked at cucumber pro and rubydoc. I then found some example blogs that don't seem up to date any more. Then I tried to go through the Github code assuming that the step definitions here would include the function calls they need and I could use those as examples.

Thanks again.
